src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -45,5 +48,31 @@ public List<IexLastTradedPrice> getLastTradedPriceForSymbols(final List<String>
     }
   }
 
+  /** Get historical prices for the symbol
+   * @param symbol a symbol to find historical prices for
+   * @param range an optional range to check
+   * @param date an optional date to check
+   */
+  public List<IexHistoricalPrices> getHistoricalPricesForSymbols(final String symbol, final String range, final String date) throws Exception{
 
+    // Input validation
+    if(StringUtils.isBlank(symbol)){ // Symbol is required
+      throw new EntityNotFoundException(symbol.getClass(), symbol);
+    } else if(StringUtils.isBlank(range) && StringUtils.isBlank(date)){ // No range or date specified
+      return iexClient.getHistoricalPricesForSymbol(symbol);
+    } else if ((StringUtils.isBlank(range) == false) && StringUtils.isBlank(date)) { // Range specified but not date
+      if(range == ""){
+        throw new HttpMessageNotReadableException("\"range\" is not allowed to be empty");
+      }
+      return iexClient.getHistoricalPricesRange(symbol, range);
+    } else if ((StringUtils.isBlank(range)) && StringUtils.isBlank(date) == false) { // Date specified but not range
```

Comment:
I think you can do isNotBlank instead of isBlank == false

Reply:
done!